### PR TITLE
Add operation classification and batch sync result tracking

### PR DESCRIPTION
**Pull request by leogdion**
## Summary
This PR introduces two new public types and a service extension to enable tracking of create vs. update operations in CloudKit batch modify requests. CloudKit's `/records/modify` endpoint doesn't indicate whether each operation created a new record or updated an existing one, so this implementation provides the documented workaround pattern.

## Key Changes

- **`OperationClassification`**: A new `Sendable`, `Equatable` struct that partitions record operations into creates and updates by comparing proposed record names against existing ones. Provides three initializers:
  - Direct comparison of proposed vs. existing record names
  - Convenience initializer that extracts names from `RecordOperation` sequences
  - Manual initializer for tests

- **`BatchSyncResult`**: A new public struct that categorizes modify response records into four buckets:
  - `created`: Records whose names were in the creates set
  - `updated`: Records whose names were in the updates set
  - `failed`: Records with `isError == true` (takes precedence over classification)
  - `unclassified`: Successful records not in either set (e.g., anonymous creates with server-assigned names)
  
  Provides computed properties for counts and totals, plus two initializers (direct array construction and classification-based partitioning).

- **`CloudKitService+Classification`**: Extension adding two helper methods:
  - `fetchExistingRecordNames(recordType:)`: Queries existing record names for a type (max 200 records per request)
  - `modifyRecords(_:classification:atomic:)`: Overload that performs a modify and returns a `BatchSyncResult` with pre-partitioned records

## Implementation Details

- Error records (identified by `recordType == "Unknown"`) are always routed to the `failed` bucket regardless of classification, preventing false positives
- The classification-based `BatchSyncResult` initializer processes records in priority order: errors first, then creates, then updates, then unclassified
- All new types are `Sendable` for async/await compatibility
- Comprehensive test coverage (5 test suites) validates partitioning logic, edge cases, and count calculations

https://claude.ai/code/session_01VxC2K7NDYP48kj3JdtYrMX

- - -
Perform an AI-assisted review on [CodePeer.com](https://codepeer.com/app/prs/github/brightdigit/MistKit/296)

### CODE REVIEW

File: Sources/MistKit/Service/CloudKitService+Classification.swift [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+  /// - Important: This issues a single `queryRecords` call with the maximum
+  ///   per-request limit of 200 records. For record types with more than 200
+  ///   records, paginate at the call site or use a custom query.
+  ///
+  /// - Parameter recordType: The CloudKit record type to scan.
+  /// - Returns: Set of existing record names.
+  /// - Throws: `CloudKitError` if the underlying query fails.
+  public func fetchExistingRecordNames(
+    recordType: String
+  ) async throws(CloudKitError) -> Set<String> {
```
Comment (leogdion) [inline Sources/MistKit/Service/CloudKitService+Classification.swift:68]: shouldn't we have an optional limit argument?

File: Sources/MistKit/Service/CloudKitService+Classification.swift [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    // Pass `limit: 200` explicitly so overload resolution picks the
+    // typed-throws variant of `queryRecords` rather than the 1-param
+    // RecordManaging-conforming overload (which has untyped throws).
+    let records = try await queryRecords(
+      recordType: recordType,
+      limit: 200
```
Comment (leogdion) [inline Sources/MistKit/Service/CloudKitService+Classification.swift]: Isn't 200 a constant somewhere? perhaps MistKit?

File: Tests/MistKitTests/PublicTypes/BatchSyncResultTests.swift [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+  /// "Unknown"; the public init does not allow forging that, so use the
+  /// minimum of one error record by setting recordType to "Unknown".
+  internal static func makeErrorRecord(name: String = "Unknown") -> RecordInfo {
+    RecordInfo(
+      recordName: name,
+      recordType: "Unknown",
+      recordChangeTag: nil,
+      fields: [:]
+    )
+  }
```
Comment (leogdion) [inline Tests/MistKitTests/PublicTypes/BatchSyncResultTests.swift]: this seems like a bad pattern. Is there an alternative?

File: Sources/MistKit/Service/CloudKitService+Classification.swift (hunk 1 above)
Comment (leogdion) [reply Sources/MistKit/Service/CloudKitService+Classification.swift:68]: Good call — added `limit: Int? = nil` to `fetchExistingRecordNames(recordType:limit:)` in 73cd1be. Defaults to CloudKit's per-request max so existing call sites don't change.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: Sources/MistKit/Service/CloudKitService+Classification.swift (hunk 2 above)
Comment (leogdion) [reply Sources/MistKit/Service/CloudKitService+Classification.swift]: Searched and there wasn't one — `200` was an inline literal in `CloudKitService+Operations.swift:104`, `+SyncOperations.swift:86`, and `+RecordManaging.swift:52`. Added `internal static let maxRecordsPerRequest: Int = 200` on `CloudKitService` in 73cd1be and reference it from `fetchExistingRecordNames`. Happy to follow up with a separate PR replacing the other three call sites if you'd like — kept the scope tight here.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: Tests/MistKitTests/PublicTypes/BatchSyncResultTests.swift (hunk 3 above)
Comment (leogdion) [reply Tests/MistKitTests/PublicTypes/BatchSyncResultTests.swift]: Agreed — the helper was reaching past the public API to forge an error record by hardcoding the `"Unknown"` sentinel.

Fixed in 73cd1be by routing through the same decoding path production uses: `RecordInfo(from: Components.Schemas.RecordResponse())`. That's the existing pattern in `RecordInfoTests.recordInfoWithUnknownRecord` (`Tests/MistKitTests/Core/RecordInfoTests.swift:12`) and produces the same shape CloudKit yields for failures, so the test no longer encodes private knowledge of the sentinel string.

The "errors win over classification" test was also updated to read the error record's `recordName` dynamically instead of asserting on `"Unknown"`.

Longer-term, if you'd like to retire the sentinel entirely (e.g. give `RecordInfo` an explicit `error: ErrorPayload?` instead of the `recordType == "Unknown"` check), happy to file that as a follow-up — felt out of scope for this PR.

---
_Generated by [Claude Code](https://claude.ai/code)_
